.github/workflows/canister-tests.yml

```diff
@@ -202,136 +202,54 @@ jobs:
         os: [ ubuntu-latest, macos-latest ]
     steps:
       - uses: actions/checkout@v3
+
+      - uses: actions/cache@v3
+        id: cache-test-archive
         with:
-          # Pull a deep copy so that we can set meaningful timestamps across builds, based on the commit history
-          fetch-depth: '0'
-
-      # Figure out which commit modified what. Each commit defined here is used as a cache layer.
-      # We use those commit shas as cache keys. Since cargo uses timestamp-based comparison,
-      # each cache layer is retrieved and timestamped when the particular commit is checked out,
-      # so that checking out a later commit tells cargo which files are newer (since git will
-      # update the mtime of newer files)
-      - name: Retrieve commit information
-        id: git_info
-        run: |
-          # Last time dependencies or toolchain were updated; used for ~/.cargo and dependencies
-          commit_deps=$(git log -1 --pretty=format:%H -- Cargo.* src/*/Cargo.toml rust-toolchain.toml)
-          echo "commit_deps=$commit_deps" >> "$GITHUB_OUTPUT"
-          echo "deps: $commit_deps"
+          path: /tmp/test-archive
+          key: ${{ runner.os }}-rust-test-archive-${{ hashFiles('src/**/*.rs', 'Cargo.*', 'src/*/*.toml', 'rust-toolchain.toml') }}
```

Comment:
Is there a better way to check for all rust (file) dependencies? I think you might get incorrect cache hits if e.g. the bundled tarball changes; for instance if it stops including an `index.js` (or rather if you don't `touch` the index.js below) you'd still get a cache hit even though in theory the tests should have been rebuilt.

Alternatively, create the dist early and add it to the hash?

Maybe I'm just worried about nothing

Reply:
The JS files are not part of the test binaries. Rather the test binaries load the `.wasm` (that has been correctly built) later. So if the JS file changes, it is ok to reuse the same test binary, because the test will see the new file once it loads the `.wasm`.

Reply:
Ah, yes! And as long as the tests are only made of pure rust files we're good. Thanks!